src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -4,56 +4,100 @@ package freechips.rocketchip.diplomacy
   def parents: Seq[LazyModule] = parent match {
     case None => Nil
     case Some(x) => x +: x.parents
   }
 
+  // [[LazyModule.scope]] stack push.
   LazyModule.scope = Some(this)
+  // Prepend this instance to [[parent]]'s list of children.
   parent.foreach(p => p.children = this :: p.children)
 
-  // suggestedName accumulates Some(names), taking the final one. Nones are ignored.
+  /** Accumulates Some(names), taking the final one. `None`s are ignored. */
   private var suggestedNameVar: Option[String] = None
+
+  /** Set instance name of [[LazyModuleImpLike]] module.
+    *
+    * @param x suggested name to [[LazyModuleImpLike]] module.
+    * */
   def suggestName(x: String): this.type = suggestName(Some(x))
   def suggestName(x: Option[String]): this.type = {
     x.foreach { n => suggestedNameVar = Some(n) }
     this
   }
 
+  /** Helper function of [[className]]. This searches upwards in the class hierarchy for the first class name that was not an anonymous scala class.*/
   private def findClassName(c: Class[_]): String = {
     val n = c.getName.split('.').last
     if (n.contains('$')) findClassName(c.getSuperclass) else n
   }
 
-  lazy val className = findClassName(getClass)
-  lazy val suggestedName = suggestedNameVar.getOrElse(className)
-  lazy val desiredName = className // + hashcode?
+  /** Get the class name of this instance. */
+  lazy val className: String = findClassName(getClass)
+  /** Set instance name. */
+  lazy val suggestedName: String = suggestedNameVar.getOrElse(className)
+  /** Set module name. */
+  lazy val desiredName: String = className // + hashcode?
 
+  /** Return instance name. */
   def name = suggestedName // className + suggestedName ++ hashcode ?
+  /** Return source line that defines this instance. */
   def line = sourceLine(info)
 
   // Accessing these names can only be done after circuit elaboration!
-  lazy val moduleName = module.name // The final Verilog Module name
+  /** Module name, used in GraphML. */
+  lazy val moduleName = module.name
+  /** Module hierarchy of this instance, used in GraphML. */
   lazy val pathName = module.pathName
-  lazy val instanceName = pathName.split('.').last // The final Verilog instance name
+  /** Instance name, used in GraphML. */
+  lazy val instanceName = pathName.split('.').last
 
+  /** [[chisel3]] hardware implementation of this [[LazyModule]],
+    * This `module` function should be declared  as `lazy val` in the subclass for lazy evaluation.
+    * Generally, it is the beginning of phase 2.
+    * */
   def module: LazyModuleImpLike
 
+  /** Generate the [[graphML]] representation for this instance. 
+    * This is a representation of the Nodes, Edges, Lazy Module hierarchy,
+    * and any other information that is added in by the implementations.
+    * It can be converted to a graphical representation of the [[LazyModule]]
+    * hierarchy with various third-party tools.
+    * Note: the name of this function should be `emitGraphML`.
```

Comment:
lol

Reply:
actually now I am not sure, maybe it really does mean `omit`?

Reply:
in which case this scaladoc is totally wrong

Reply:
Oh wait, I think you're right. Let's delete this last sentence, then.

Reply:
```suggestion
```